polynote-kernel/src/main/scala/polynote/kernel/Kernel.scala

```diff
@@ -29,6 +29,11 @@ trait Kernel {
     */
   def parametersAt(id: CellID, pos: Int): TaskC[Option[Signatures]]
 
+  /**
+    * Provide a location for the definition of whatever is at the given position in the given cell
+    */
+  def goToDefinition(id: Either[String, CellID], pos: Int): TaskC[(List[DefinitionLocation], Option[String])]
```

Comment:
What is the `String` in `id`? The source file name in case you go to definition inside a file browser?

Might it be worth creating a type alias for this (off the top of my head, maybe `type sourceOrCell`) for readability?

Reply:
Oh I see you called it `cellOrFile` in some of the UI code, that'd work fine too (I guess "Cell" first because it's "Right" in the Either?